### PR TITLE
Toggle completion menu

### PR DESCRIPTION
**Pull request by Sam-programs**
This pr gives the user the ability to hide all the cmp, by wrapping every `nvim_win*` function needed and storing the results before hiding the windows.
Fix:
https://github.com/hrsh7th/nvim-cmp/issues/1698
https://github.com/hrsh7th/nvim-cmp/issues/1013

### CODE REVIEW

**Comment by dsully** [comment]
Would love to see this merged.

File: lua/cmp/view/ghost_text_view.lua [hunk 1]
```diff
@@ -26,9 +26,13 @@ ghost_text_view.new = function()
...
-      return win == self.win
-    end,
-    on_line = function(_, _, _, on_row)
+      if self.extmark_id then
+        vim.api.nvim_buf_del_extmark(self.extmark_buf, ghost_text_view.ns, self.extmark_id)
+        self.extmark_id = nil
+      end
+      if win ~= self.win then
+        return
+      end
```
Comment (Sam-programs) [inline lua/cmp/view/ghost_text_view.lua:35]: This temporary because inline ghost text was broken in cmp when this was forked.

**Comment by Sam-programs** [comment]
This still needs documentation, but i am not sure if the current config interface is good, i added 2 options.
```lua
window = {
      auto_hide = false,
      show_on_cursor_update = false,
}
```
`auto_hide` hides the windows after they opened until `require('cmp.utils.window').show_all()` is called or when the cursor moves in the menu if `show_on_cursor_update` is true.
However `show_on_cursor_update` doesn't work well because the cursor doesn't move with a pre-selected item,

**Comment by sleepy-day** [comment]
Thanks for the work on this branch, I've been using it for a couple of months and forgot to check if it got merged in but it looks like it hasn't. Would it be possible to get this reviewed and merged in? Might need more work since it looks like it has merge conflicts and not sure if the creator of the branch is still around but if they're not I can clean up any issues and resubmit. Thank you.

**Comment by Sam-programs** [comment]
> not sure if the creator of the branch is still around but if they're not I can clean up any issues

I am still here, but I don't plan to work on this pr anymore because I don't use cmp anymore. Sorry.

**Pull request closed without merging**